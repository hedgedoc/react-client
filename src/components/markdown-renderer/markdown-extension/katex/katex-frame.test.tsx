/*
 * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
 *
 * SPDX-License-Identifier: AGPL-3.0-only
 */

import { render } from '@testing-library/react'
import KatexFrame from './katex-frame'

describe('katex frame', () => {
  describe('renders a valid latex expression', () => {
    it('as implicit inline', () => {
      const view = render(<KatexFrame expression={'\\int_0^\\infty x^2 dx'}></KatexFrame>)
      expect(view.container).toMatchSnapshot()
    })
    it('as explicit inline', () => {
      const view = render(<KatexFrame expression={'\\int_0^\\infty x^2 dx'} block={false}></KatexFrame>)
      expect(view.container).toMatchSnapshot()
    })
    it('as explicit block', () => {
      const view = render(<KatexFrame expression={'\\int_0^\\infty x^2 dx'} block={true}></KatexFrame>)
      expect(view.container).toMatchSnapshot()
    })
  })

  describe('renders an error for an invalid latex expression', () => {
    it('as implicit inline', () => {
      const view = render(<KatexFrame expression={'\\alf'}></KatexFrame>)
      expect(view.container).toMatchSnapshot()
    })
    it('as explicit inline', () => {
      const view = render(<KatexFrame expression={'\\alf'} block={false}></KatexFrame>)
      expect(view.container).toMatchSnapshot()
    })
    it('as explicit block', () => {
      const view = render(<KatexFrame expression={'\\alf'} block={true}></KatexFrame>)
      expect(view.container).toMatchSnapshot()
    })
  })
})
