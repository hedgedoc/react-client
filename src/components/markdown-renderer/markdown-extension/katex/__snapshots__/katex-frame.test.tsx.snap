// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`katex frame renders a valid latex expression as explicit block 1`] = `
<div>
  <div
    data-testid="katex-block"
  >
    <span
      class="katex-display"
    >
      <span
        class="katex"
      >
        <span
          class="katex-mathml"
        >
          <math
            display="block"
            xmlns="http://www.w3.org/1998/Math/MathML"
          >
            <semantics>
              <mrow>
                <msubsup>
                  <mo>
                    ∫
                  </mo>
                  <mn>
                    0
                  </mn>
                  <mi
                    mathvariant="normal"
                  >
                    ∞
                  </mi>
                </msubsup>
                <msup>
                  <mi>
                    x
                  </mi>
                  <mn>
                    2
                  </mn>
                </msup>
                <mi>
                  d
                </mi>
                <mi>
                  x
                </mi>
              </mrow>
              <annotation
                encoding="application/x-tex"
              >
                \\int_0^\\infty x^2 dx
              </annotation>
            </semantics>
          </math>
        </span>
        <span
          aria-hidden="true"
          class="katex-html"
        >
          <span
            class="base"
          >
            <span
              class="strut"
              style="height: 2.3262em; vertical-align: -0.9119em;"
            />
            <span
              class="mop"
            >
              <span
                class="mop op-symbol large-op"
                style="margin-right: 0.44445em; position: relative; top: -0.0011em;"
              >
                ∫
              </span>
              <span
                class="msupsub"
              >
                <span
                  class="vlist-t vlist-t2"
                >
                  <span
                    class="vlist-r"
                  >
                    <span
                      class="vlist"
                      style="height: 1.4143em;"
                    >
                      <span
                        style="top: -1.7881em; margin-left: -0.4445em; margin-right: 0.05em;"
                      >
                        <span
                          class="pstrut"
                          style="height: 2.7em;"
                        />
                        <span
                          class="sizing reset-size6 size3 mtight"
                        >
                          <span
                            class="mord mtight"
                          >
                            0
                          </span>
                        </span>
                      </span>
                      <span
                        style="top: -3.8129em; margin-right: 0.05em;"
                      >
                        <span
                          class="pstrut"
                          style="height: 2.7em;"
                        />
                        <span
                          class="sizing reset-size6 size3 mtight"
                        >
                          <span
                            class="mord mtight"
                          >
                            ∞
                          </span>
                        </span>
                      </span>
                    </span>
                    <span
                      class="vlist-s"
                    >
                      ​
                    </span>
                  </span>
                  <span
                    class="vlist-r"
                  >
                    <span
                      class="vlist"
                      style="height: 0.9119em;"
                    >
                      <span />
                    </span>
                  </span>
                </span>
              </span>
            </span>
            <span
              class="mspace"
              style="margin-right: 0.1667em;"
            />
            <span
              class="mord"
            >
              <span
                class="mord mathnormal"
              >
                x
              </span>
              <span
                class="msupsub"
              >
                <span
                  class="vlist-t"
                >
                  <span
                    class="vlist-r"
                  >
                    <span
                      class="vlist"
                      style="height: 0.8641em;"
                    >
                      <span
                        style="top: -3.113em; margin-right: 0.05em;"
                      >
                        <span
                          class="pstrut"
                          style="height: 2.7em;"
                        />
                        <span
                          class="sizing reset-size6 size3 mtight"
                        >
                          <span
                            class="mord mtight"
                          >
                            2
                          </span>
                        </span>
                      </span>
                    </span>
                  </span>
                </span>
              </span>
            </span>
            <span
              class="mord mathnormal"
            >
              d
            </span>
            <span
              class="mord mathnormal"
            >
              x
            </span>
          </span>
        </span>
      </span>
    </span>
  </div>
</div>
`;

exports[`katex frame renders a valid latex expression as explicit inline 1`] = `
<div>
  <span
    data-testid="katex-inline"
  >
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <msubsup>
                <mo>
                  ∫
                </mo>
                <mn>
                  0
                </mn>
                <mi
                  mathvariant="normal"
                >
                  ∞
                </mi>
              </msubsup>
              <msup>
                <mi>
                  x
                </mi>
                <mn>
                  2
                </mn>
              </msup>
              <mi>
                d
              </mi>
              <mi>
                x
              </mi>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              \\int_0^\\infty x^2 dx
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height: 1.2151em; vertical-align: -0.3558em;"
          />
          <span
            class="mop"
          >
            <span
              class="mop op-symbol small-op"
              style="margin-right: 0.19445em; position: relative; top: -0.0006em;"
            >
              ∫
            </span>
            <span
              class="msupsub"
            >
              <span
                class="vlist-t vlist-t2"
              >
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height: 0.8593em;"
                  >
                    <span
                      style="top: -2.3442em; margin-left: -0.1945em; margin-right: 0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height: 2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          0
                        </span>
                      </span>
                    </span>
                    <span
                      style="top: -3.2579em; margin-right: 0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height: 2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          ∞
                        </span>
                      </span>
                    </span>
                  </span>
                  <span
                    class="vlist-s"
                  >
                    ​
                  </span>
                </span>
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height: 0.3558em;"
                  >
                    <span />
                  </span>
                </span>
              </span>
            </span>
          </span>
          <span
            class="mspace"
            style="margin-right: 0.1667em;"
          />
          <span
            class="mord"
          >
            <span
              class="mord mathnormal"
            >
              x
            </span>
            <span
              class="msupsub"
            >
              <span
                class="vlist-t"
              >
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height: 0.8141em;"
                  >
                    <span
                      style="top: -3.063em; margin-right: 0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height: 2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          2
                        </span>
                      </span>
                    </span>
                  </span>
                </span>
              </span>
            </span>
          </span>
          <span
            class="mord mathnormal"
          >
            d
          </span>
          <span
            class="mord mathnormal"
          >
            x
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`katex frame renders a valid latex expression as implicit inline 1`] = `
<div>
  <span
    data-testid="katex-inline"
  >
    <span
      class="katex"
    >
      <span
        class="katex-mathml"
      >
        <math
          xmlns="http://www.w3.org/1998/Math/MathML"
        >
          <semantics>
            <mrow>
              <msubsup>
                <mo>
                  ∫
                </mo>
                <mn>
                  0
                </mn>
                <mi
                  mathvariant="normal"
                >
                  ∞
                </mi>
              </msubsup>
              <msup>
                <mi>
                  x
                </mi>
                <mn>
                  2
                </mn>
              </msup>
              <mi>
                d
              </mi>
              <mi>
                x
              </mi>
            </mrow>
            <annotation
              encoding="application/x-tex"
            >
              \\int_0^\\infty x^2 dx
            </annotation>
          </semantics>
        </math>
      </span>
      <span
        aria-hidden="true"
        class="katex-html"
      >
        <span
          class="base"
        >
          <span
            class="strut"
            style="height: 1.2151em; vertical-align: -0.3558em;"
          />
          <span
            class="mop"
          >
            <span
              class="mop op-symbol small-op"
              style="margin-right: 0.19445em; position: relative; top: -0.0006em;"
            >
              ∫
            </span>
            <span
              class="msupsub"
            >
              <span
                class="vlist-t vlist-t2"
              >
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height: 0.8593em;"
                  >
                    <span
                      style="top: -2.3442em; margin-left: -0.1945em; margin-right: 0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height: 2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          0
                        </span>
                      </span>
                    </span>
                    <span
                      style="top: -3.2579em; margin-right: 0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height: 2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          ∞
                        </span>
                      </span>
                    </span>
                  </span>
                  <span
                    class="vlist-s"
                  >
                    ​
                  </span>
                </span>
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height: 0.3558em;"
                  >
                    <span />
                  </span>
                </span>
              </span>
            </span>
          </span>
          <span
            class="mspace"
            style="margin-right: 0.1667em;"
          />
          <span
            class="mord"
          >
            <span
              class="mord mathnormal"
            >
              x
            </span>
            <span
              class="msupsub"
            >
              <span
                class="vlist-t"
              >
                <span
                  class="vlist-r"
                >
                  <span
                    class="vlist"
                    style="height: 0.8141em;"
                  >
                    <span
                      style="top: -3.063em; margin-right: 0.05em;"
                    >
                      <span
                        class="pstrut"
                        style="height: 2.7em;"
                      />
                      <span
                        class="sizing reset-size6 size3 mtight"
                      >
                        <span
                          class="mord mtight"
                        >
                          2
                        </span>
                      </span>
                    </span>
                  </span>
                </span>
              </span>
            </span>
          </span>
          <span
            class="mord mathnormal"
          >
            d
          </span>
          <span
            class="mord mathnormal"
          >
            x
          </span>
        </span>
      </span>
    </span>
  </span>
</div>
`;

exports[`katex frame renders an error for an invalid latex expression as explicit block 1`] = `
<div>
  <div
    data-testid="katex-block"
  >
    <div
      class="fade alert alert-danger show"
      role="alert"
    >
      KaTeX parse error: Undefined control sequence: \\alf at position 1: \\̲a̲l̲f̲
    </div>
  </div>
</div>
`;

exports[`katex frame renders an error for an invalid latex expression as explicit inline 1`] = `
<div>
  <span
    data-testid="katex-inline"
  >
    <div
      class="fade d-inline-block alert alert-danger show"
      role="alert"
    >
      KaTeX parse error: Undefined control sequence: \\alf at position 1: \\̲a̲l̲f̲
    </div>
  </span>
</div>
`;

exports[`katex frame renders an error for an invalid latex expression as implicit inline 1`] = `
<div>
  <span
    data-testid="katex-inline"
  >
    <div
      class="fade d-inline-block alert alert-danger show"
      role="alert"
    >
      KaTeX parse error: Undefined control sequence: \\alf at position 1: \\̲a̲l̲f̲
    </div>
  </span>
</div>
`;
